pkg/operators/wasm/maps.go

```diff
@@ -213,3 +276,43 @@ func (i *wasmOperatorInstance) mapDelete(ctx context.Context, m wapi.Module, sta
 	}
 	stack[0] = 0
 }
+
+// mapRelease close the map and release the handle.
+// Params:
+// - stack[0]: Map handle
+// Return value:
+// - 0 on success, 1 on error
+func (i *wasmOperatorInstance) mapRelease(ctx context.Context, m wapi.Module, stack []uint64) {
+	mapHandle := wapi.DecodeU32(stack[0])
+
+	i.createdMapMutex.RLock()
+	_, ok := i.createdMap[mapHandle]
+	i.createdMapMutex.RUnlock()
+
+	if !ok {
+		i.logger.Warnf("mapRelease: map %d was not created by newMap() or was already closed", mapHandle)
```

Comment:
Shouldn't this test be done after `getHandle[*ebpf.Map]()`? So if the wasm code calls mapRelease() on a handle of different type, it does not say "map %d".